### ISSUE TITLE
Error Not Being Raised for Non Existent Zip

### ISSUE THREAD

**Issue by cancan101** [plrthink/react-native-zip-archive#3]
Tis is calling the `then` rather than `catch` where the specified file does not exist:
```js
   ZipArchive.unzip(RNFS.CachesDirectoryPath + '/XXXXX.zip', RNFS.CachesDirectoryPath + '/YYYY')
      .then((o) => {
        alert('o: ' + o)
      })
      .catch((error) => {
        alert('E: ' + error)
      });
```

**Comment by cancan101** [comment]
The same seems to happen with the file exists but is not a valid zip.

**Comment by plrthink** [comment]
@cancan101 I will fix this ASAP.

**Comment by WestleyArgentum** [comment]
Also seeing this problem...

But even when the file exists and is a valid zip, this doesn't seem to work... any tips on debugging?

**Comment by plrthink** [comment]
@WestleyArgentum I'm using `react-native-fs` to get access to the zip file on the simulator to check the it. I think any method which can get path to the zip file is ok.

I would improve this.

**Comment by WestleyArgentum** [comment]
I'm using `react-native-fs` to download the zip, and I use `RNFS.readDir` to check and make sure the zip file is downloaded correctly... but then when I try to unzip it nothing happens and I don't see any errors...

**Comment by plrthink** [comment]
@WestleyArgentum Your mean the promise is hanging?

**Comment by plrthink** [comment]
@WestleyArgentum I think maybe you're dong something wrong wth promise code, e.g. not return the promise from the inner promise chain.

**Comment by WestleyArgentum** [comment]
Sorry I disappeared there, but in this case I don't think the error is promise related -- my code is just like the example:

```
    ZipArchive.unzip(RNFS.DocumentDirectoryPath + '/images.zip', RNFS.DocumentDirectoryPath)
            .then(() => {
                // check to see what exists in the directory...
                RNFS.readDir(RNFS.DocumentDirectoryPath)
                  .then((result) => {
                    // shows images.zip, but not an unzipped folder or any of the contents of the zip...
                    console.log('things in directory:', result);
                });
            })
            .catch((err) => {
                cb(err);
            });
```

Does that look ok?

**Comment by WestleyArgentum** [comment]
Ah, I realized I can use xcode to step through the code and my problem was with directories. Would be really cool to have those errors raised through `.catch` though

Thanks for your help!

**Comment by plrthink** [comment]
@WestleyArgentum oh, can you explain a little bit about your resolution? So I can collect errors should be raised.

**Comment by WestleyArgentum** [comment]
So, I've never spent much time in objective c, but I think the problem is here: https://github.com/plrthink/react-native-zip-archive/blob/master/RNZipArchive.m#L19

It looks like these methods take in a pointer to a pointer to an error: https://github.com/plrthink/react-native-zip-archive/blob/fb47345d3094e7e85bf49af747758acea7fe43d3/SSZipArchive/SSZipArchive.m#L47

And that you should pass that in, check if it gets set, and raise an error instead of calling the callback

**Comment by plrthink** [comment]
fixed with b622cb07150ee75b61d71868a33751751da79f13

**Comment by WestleyArgentum** [comment]
nice, thanks!
